wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointMonitorImpl.java

```diff
@@ -213,8 +216,24 @@ public void monitor() {
     }
   }
 
+  protected void sleep(long durationNs) throws InterruptedException {
+    synchronized (this.refreshRequired) {
+      this.refreshRequired.wait(TimeUnit.NANOSECONDS.toMillis(durationNs));
```

Comment:
This `wait()` may return earlier than specified time when the value of `this.refreshRequired` hasn't changed. You need a loop to continue waiting. `ClusterTopologyMonitorImpl` has good implementation of `delay()` with a loop.